core/textile/sync/restore.go

```diff
@@ -0,0 +1,40 @@
+package sync
+
+import (
+	"context"
+
+	"github.com/FleekHQ/space-daemon/log"
+)
+
+// replicate a local thread on the hub
+func (s *synchronizer) restoreBucket(ctx context.Context, bucket string) error {
+
+	b, err := s.getBucket(ctx, bucket)
+	if err != nil {
+		log.Error("Error in getBucket", err)
+		return err
+	}
+
+	dir, err := b.ListDirectory(ctx, "")
+	if err != nil {
+		log.Error("Error in ListDirectory", err)
+		return err
+	}
+
+	dirPaths := make([]string, 0)
+	for _, item := range dir.Item.Items {
+		dirPaths = append(dirPaths, item.Path)
+	}
+
+	mirrorFiles, err := s.model.FindMirrorFileByPaths(ctx, dirPaths)
```

Comment:
This will make sure any existing files are updated to the latest version.  But we still need something to browse remote bucket and add any missing files to the local bucket right? Maybe that's the other piece we talked about in standup that needs to be done.

Reply:
@jsonsivar 

Yes, that's the meat now: https://github.com/FleekHQ/space-daemon/pull/207/files#diff-e4729a5c7a413a5b91b5b88b329f166525e3b3daec3b74e4991f7a19e1165c70R11

It needs very extensive testing but:

 * it relies only on the buckets now,
 * it uses the newly discovered `Item.Metadata.UpdatedAt`.

Ideally, the same logic should follow in https://github.com/FleekHQ/space-daemon/pull/207/files#diff-0fdaaca29057f19da1dabe9fdd5aa736707865612957999372f7ea54648e8993R218 but I'd wait for #207 

Also, I think that the newly added `bucket.Each()` could be used in many other places (we iterate over buckets everywhere) as well as `synchronizer.processBucketBackupOn()` should be smarter but, again, I'd wait for #207 and consider `bucket.Sync()` using the CID differences.

Reply:
Okay sounds good.  I would also triple check to make sure the `Metadata.UpdatedAt` is actually for the file and not the `roles` metadata. E.g., if someone updates the file but doesn't change the ACL would this get updated?